src/chapters/sql-part2/studio.rst

```diff
@@ -0,0 +1,184 @@
+      last_name VARCHAR(255),
+      loan_id INT
+   );
+
+To fill the table, use the ``patrons.csv`` file.
+
+The ``reference_books`` table
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Use the following query to create the ``reference_books`` table.
+
+.. sourcecode:: mysql
+   :linenos:
+
+   CREATE TABLE reference_books (
+	  reference_id INT AUTO_INCREMENT PRIMARY KEY,
+      edition INT,
+      book_id INT,
+      FOREIGN KEY (book_id)
+	 	 REFERENCES book(book_id)
+         ON UPDATE SET NULL
+         ON DELETE SET NULL
+   );
+
+To fill the table, use the following query.
+
+.. sourcecode:: mysql
+   :linenos:
+
+   INSERT INTO reference_books(edition, book_id)
+   VALUE (5,32);
+
+The ``genre`` table
+^^^^^^^^^^^^^^^^^^^
+
+Use the following query to create the ``genre`` table.
+
+.. sourcecode:: mysql
+   :linenos:
+
+   CREATE TABLE genre (
+	 genre_id INT AUTO_INCREMENT PRIMARY KEY,
+     genre_type VARCHAR(100)
+   );
+
+Use the ``genres.csv`` file to fill the ``genre`` table.
+
+The ``loan`` table
+^^^^^^^^^^^^^^^^^^
+
+Use the following query to create the ``loan`` table.
+
+.. sourcecode:: mysql
+   :linenos:
+
+   CREATE TABLE loan (
+	  loan_id INT AUTO_INCREMENT PRIMARY KEY,
+      patron_id INT,
+      date_out DATE,
+      date_in DATE,
+      book_id INT,
+      FOREIGN KEY (book_id)
+		 REFERENCES book(book_id)
+         ON UPDATE SET NULL
+		 ON DELETE SET NULL
+   );
+
+For now, the ``loan`` table can stay empty! We will add information to it in a bit!
+
+Warm-up Queries
+---------------
+
+Write the following 2 queries to get warmed up.
+
+#. Return the reference book titles and their ISBNs.
+#. Return all of the books written by authors who are currently living.
+
+Loan Out a Book
+---------------
+
+A big function that you need to implement for the library is a script that updates the database when a book is loaned out.
+
+The script needs to perform the following functions:
+
+#. Change ``available`` to ``FALSE`` for the appropriate book.
+#. Add a new row to the ``loan`` table with today's date as the ``date_out`` and the ids in the row matching the appropriate ``patron_id`` and ``book_id``.
+#. Update the appropriate ``patron`` with the ``loan_id`` for the new row created in the ``loan`` table.
+
+You can use any patron and book that strikes your fancy to create this script!
+
+Check a Book Back In
+--------------------
+
+Working with the same patron and book, create the new script!
+
+The other key function that we need to implement is checking a book back in.
+To do so, the script needs to:
+
+#. Change ``available`` to ``TRUE`` for the appropriate book.
+#. Update the appropriate row in the ``loan`` table with today's date as the ``date_in``.
+#. Update the appropriate ``patron`` changing ``loan_id`` back to ``null``.
+
+Wrap-up Query
+-------------
+
+Write a query to wrap up the studio. This query should return the names of the patrons with the genre of every book they currently have checked out.
```

Comment:
Since the students only altered one patron/loan/book in the previous two sections, will this query return any results?  If the students check for books currently out by looking in ``patron``, they will find all of the ``loan_id`` values set to ``NULL``.

Did you want the students to request records from ``loan`` and use a join or subquery with ``genre`` to create the result set, or should the students create several checkouts before trying the wrap-up?

Reply:
Fixed! They will have to get the genres from the books joined with loans joined with patrons.